packages/eslint-config/lib/index.js

```diff
@@ -0,0 +1,81 @@
+const path = require('path');
+
+const config = {
+	env: {
+		browser: true,
+		es6: true,
+		mocha: true,
+		node: true
+	},
+	parserOptions: {
+		ecmaVersion: 'latest',
+		sourceType: 'module'
+	},
+	rules: {
+		RuleToAdd: 'RuleToAdd'
+	},
+	globals: {
+		cy: true,
+		cypress: true,
+		Cypress: true,
+		fetch: true,
+		requireText: true
+	},
+	plugins: ['no-only-tests'],
+	extends: [],
+	overrides: [
+		{
+			files: ['test/**/*.js', 'tests/**/*.js', 'cypress/**/*.js'],
+			rules: {
+				'no-only-tests/no-only-tests': 'error'
+			}
+		}
+	],
+	settings: {}
+};
+
+const packageJson = require(path.join(process.cwd(), 'package.json'));
+
+const packageJsonContainsPackage = (packageName) => {
+	const { dependencies, devDependencies, peerDependencies } = packageJson;
+	return (
+		(dependencies && dependencies[packageName]) ||
+		(devDependencies && devDependencies[packageName]) ||
+		(peerDependencies && peerDependencies[packageName])
+	);
+};
+
+if (
+	packageJsonContainsPackage('react') ||
+	packageJsonContainsPackage('preact')
+) {
+	config.plugins.push('react');
+	config.extends.push('plugin:react/recommended');
+
+	Object.assign(config.rules, {
+		'react/display-name': 'off',
+		'react/prop-types': 'off',
+		'react/no-danger': 'off',
+		'react/no-render-return-value': 'off'
+	});
+
+	Object.assign(config.settings, {
+		react: {
+			version: 'detect'
+		}
+	});
+}
+
+if (packageJsonContainsPackage('jest')) {
+	config.env.jest = true;
+}
+
+if (packageJson && packageJson.eslintConfig) {
+	Object.assign(config.env, packageJson.eslintConfig.env);
+	Object.assign(config.parserOptions, packageJson.eslintConfig.parserOptions);
+	Object.assign(config.rules, packageJson.eslintConfig.rules);
+	Object.assign(config.globals, packageJson.eslintConfig.globals);
+	Object.assign(config.settings, packageJson.eslintConfig.settings);
+}
+
```

Comment:
**suggestion:** I don't think we actually need any of this, as this stuff is handled by eslint-config-next. We can probably simplify a lot by deferring React/Preact/Jest decisions to them.

Reply:
So I umm'd and ahh'd on this too and thought of a scenario where an app only installs our reliability-kit-eslint-config and didn't already have the next-eslint-config. Would this block of code need to be present then or not?

Reply:
I think we shouldn't duplicate the code across configs, and I don't think we've looked into React / JSX usage enough to make decisions about how we lint it yet, whereas the core rules we've gone thorough and made decisions about. If I'm honest I don't like this code anyway and think we should remove from next-eslint-config at a later date 😁 maybe having a separate config for React-based apps.